subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -1375,7 +1408,9 @@ static int lwm2m_rd_client_init(const struct device *dev)
 {
 	client.ctx = NULL;
 	client.rd_message.ctx = NULL;
+	k_mutex_lock(&state_lock, K_FOREVER);
 	client.engine_state = ENGINE_IDLE;
+	k_mutex_unlock(&state_lock);
 	k_mutex_init(&client.mutex);
```

Comment:
If client instance have mutex already, why we need another one for state variable, that is inside the client context?

I think we should use the client.mutex on all places.

Reply:
Agree